### DISCUSSION TITLE
ESP8266 Compatibility

### DISCUSSION THREAD

**Discussion by iKrushYou** [HomeSpan/HomeSpan#5] [General]
Hey, I only see mention of ESP32 so I don't want to make any assumptions about compatibility. I already have a bunch of ESP8266s and from what i've read they're more or less the same just with different specs

**Comment by HomeSpan** [comment]
I've not tried compiling for an ESP8266.  To the extent that the Arduino-ESP8266 library is the same as the Arduino-ESP32 library, it would likely compile.  However, I'm not sure about performance, or if there are other material differences that would inhibit HomeSpan from operating properly.  Have you given it a try?

**Reply by iKrushYou** [reply]
I have not giving it a try yet I only just discovered this library today. I will probably give it a try over the weekend if I get some time

**Reply by nimartin28** [reply]
Any further thoughts on this? I love the library--it's terrific, congratulations. But I've found the ESP-32 to be less stable than the ESP-8266 so a port would be great. (My latest problems with the ESP-32 center around WiFi connectivity. Nothing that hasn't been reported in a million different posts about the ESP-32, but no one seems to have a reliable solution. I'm using the KS0413 keyestudio ESP32 Core Board, which is based on an ESP-32-WROOM.)

**Reply by nimartin28** [reply]
(I don't mean to congratulate the wrong person for creating the library--pushed buttons too quickly.)

**Reply by DesignLowKey** [reply]
I've tried uploading the same code to an ESP8266 DEV Board... Here's the error-code...
c:\Users\Documents\Arduino\libraries\HomeSpan\src/HomeSpan.h:31:2: error: #error ERROR: HOMESPAN IS ONLY AVAILABLE FOR ESP32 MICROCONTROLLERS!

I don't think it's that easy^^

**Reply by HomeSpan** [reply]
No, HomeSpan is not compatible with the 8266, though an 8266 can be used as a remote sensor using HomeSpan's SpanPoint functionality.

**Comment by HomeSpan** [comment]
I've not had any problems with my WiFi on the ESP32, which perhaps is ironic because I picked the ESP32 after having lots of problems with various WiFi boards designed for the Arduino.  My thinking at the time was that the dual-core structure of some of the ESP32 chips (though not all are dual core) would provide the stability run the WiFi stack in one core and all my HomeSpan code in the other.  I'll look into acquiring an 8266 and see what might be required to port.

**Reply by HerbOtto** [reply]
I am starting to develop connectivity issues. The MQTT service (also running with homespan) detects the WiFi dropout and reconnects however the homespan service remains offline. Could be a software issue since the hardware stack is in-tact?

I'm using the AITRIP ESP-WROOM-32 - started testing on another board today. Which board are you using?

I tried another library a while back on the 8266 and it worked provided you ran the chip at 160 MHz. So I know it's possible. The number of devices it can handle remains the question. Converting the code may require Ticker for the Blinker class and convert nvs_flash to EEPROM calls.

**Reply by HomeSpan** [reply]
I'm using an ESP32-WROOM-32 from [Adafruit](https://www.adafruit.com/product/3405) and the WiFi has been rock steady so far.  I just bought a few more from Adafruit to build some more devices, and their latest batch uses the ESP32-WROOM-32E, which seems to be the direct successor of the WROOM-32.

**Reply by nimartin28** [reply]
An ESP8266 version would be exciting because it seems like a lot of HomeKit projects don't need nearly as many ports as the '32 offers.  I still haven't figured out where my problem is: my previous information that I'm using a KS0413 key studio ESP32 Core Board is almost certainly wrong as the serial ports are not on the right pin-outs for that board. My information was based on a phone call to MicroCenter, who rebrands their products in the stores--and doesn't seem to know what they have. It is, however, almost certainly an Espressif-made ESP32-WROOM-32 since that I can read on the chip.

**Comment by prof-miles0** [comment]
Hi! This is a fantastic library for the ESP32. But has anyone tried to use it on ESP8266 chips? I would like to develop some small projects like switches or temperature sensors on ESP-01S as I only need a few ports.

Should the code be modified in any way to properly work on this kind of devices or it would never work because of the specs?

**Reply by HomeSpan** [reply]
Unfortunately, the ESP8266 is sufficiently different from the ESP32 that a port would require a substantial amount of re-work.  I've not explored whether all the libraries used by HomeSpan are part of the ESP8266 code base.

**Reply by danilkorotkov** [reply]
Esp8266 is dead and stopped developing . It was like the first step for stupid smart base things to turn on/off something: outlets, lights with a small number of gpio lines or to add brains for 8bit MCU’s via building Serial protocol.    For your simple old devices there are Maxim Kulkin’s examples and Docker, Ravencore... Most of vendors (example tuya) started to use esp32 or RTL chips, so if you are build something new for your smart home use esp32 instead of esp8266.

**Reply by nimartin28** [reply]
I do use Kulkin's library, but in my opinion, this one is better. The 8266, quirky and basic as it is, can fit into really tight spaces. And I know I'm not alone in still using it given how many are being sold. I'm a serious hobbyist--my tongue-in-cheek definition of which is: I'm not afraid of data sheets or miscellaneous parts, but I need to buy in quantities of less than 3,000 units. So far I haven't seen a version of the ESP32--even the C3 mini--that isn't soldered onto a breakout board that makes it huge and costs substantially more than the 8266. Do you know of an ESP32 vendor/package that is as small as an 8266? Thanks.

**Reply by danilkorotkov** [reply]
HomeSpan is better because esp32 has more resources and espressif supports this chip better for Arduino community.

**Reply by nimartin28** [reply]
I was asking a genuine question, not trying to get into a competition about who is right about which chip is better. Do you know of a physically small version of the ESP-32 that can be bought in small quantities?

**Reply by danilkorotkov** [reply]
So you can’t buy esp32 and want to get esp8266 support here for free?) fork and try to refactor WiFi, mdns, crypto and storage.  if you so “serious hobbyist” it will be easy for you)

**Reply by nimartin28** [reply]
Such an interesting series of exchanges, thank you. And since tone is hard to read online, I’ll add that I mean that sincerely: you’ve given me some ideas for my real job, which gets into how people communicate online. I’m now terminating our conversation: I’ve unsubscribed to notifications with this thread and don’t plan to check it again, as I can’t give you either the self-affirmation or the fight you are looking for.

**Reply by danilkorotkov** [reply]
Two years ago I asked Maxim Kulkin why did he stopped to improve his library. You can ask too.
